### PR TITLE
XS-128 improve drawer design

### PR DESCRIPTION
**Pull request by SemenStruchev**

### CODE REVIEW

File: src/components/Drawer.jsx [hunk 1]
```diff
@@ -13,86 +13,82 @@ const sizes = {
...
-                        className={clsx(
-                            "fixed inset-y-0 flex max-w-full",
-                            position === "right" ? "right-0" : "left-0",
-                            classNames.dialogContent,
-                        )}
-                    >
+export const Drawer = forwardRef(
+    ({ isOpen = false, title, size = "medium", content, onClose, classNames = {}, position = "right" }, ref) => {
+        return (
+            <Transition.Root show={isOpen} as={Fragment} ref={ref}>
```
Comment (manojx031) [inline src/components/Drawer.jsx:19]: resolve lint errors

**Review by manojx031 — CHANGES_REQUESTED** [review]

File: src/components/Drawer.jsx [hunk 2]
```diff
@@ -13,86 +13,82 @@ const sizes = {
...
-                            leaveFrom="translate-x-0"
-                            leaveTo={position === "right" ? "translate-x-full" : "-translate-x-full"}
+                            enter="ease-in-out duration-500"
+                            enterFrom="opacity-0"
+                            enterTo="opacity-100"
+                            leave="ease-in-out duration-500"
+                            leaveFrom="opacity-100"
+                            leaveTo="opacity-0"
+                        >
+                            <Dialog.Overlay className="absolute inset-0 bg-black bg-opacity-80 transition-opacity" />
```
Comment (rushi) [inline src/components/Drawer.jsx:36]: Can you add a `classNames` prop for this like we have `classNames.content` below.

Also please provide a screenshot of how the updated Drawer looks

**Review by rushi — CHANGES_REQUESTED** [review]

File: src/components/Drawer.jsx [hunk 3]
```diff
@@ -102,10 +98,16 @@ const CloseButton = ({ onClose }) => {
...
-    content: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
+    content: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
     onClose: PropTypes.func.isRequired,
     classNames: PropTypes.object,
     position: PropTypes.string,
 };
+
+Drawer.defaultProps = {
+    title: "",
+    classNames: "",
```
Comment (manojx031) [inline src/components/Drawer.jsx:111]: classNames is object above, but you are passing string here. change this to empty object. Also, these values are already declared above, why do we need to redeclare here ?

File: src/components/Drawer.jsx [hunk 4]
```diff
@@ -102,10 +98,16 @@ const CloseButton = ({ onClose }) => {
 };
 
 Drawer.propTypes = {
-    isOpen: PropTypes.bool,
+    isOpen: PropTypes.bool.isRequired,
     title: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
-    content: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
+    content: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
```
Comment (manojx031) [inline src/components/Drawer.jsx:103]: why is this change required ?

File: src/components/Drawer.jsx [hunk 5]
```diff
@@ -13,86 +13,82 @@ const sizes = {
...
-                        <Dialog.Overlay className="absolute inset-0 bg-gray bg-opacity-75 transition-opacity" />
-                    </Transition.Child>
-                    <div
-                        className={clsx(
-                            "fixed inset-y-0 flex max-w-full",
-                            position === "right" ? "right-0" : "left-0",
-                            classNames.dialogContent,
-                        )}
-                    >
+export const Drawer = forwardRef(
```
Comment (manojx031) [inline src/components/Drawer.jsx:16]: why is forwardRef needed ?

File: src/components/Drawer.jsx (hunk 2 above)
Comment (manojx031) [inline src/components/Drawer.jsx:36]: As per the figma, we need to implement this for modals as well. can you make the changes for modals as well ? Also, once the change is done please post the screenshot of the fix for modal and slider

**Review by manojx031 — CHANGES_REQUESTED** [review]

File: src/components/Drawer.jsx (hunk 2 above)
Comment (SemenStruchev) [reply src/components/Drawer.jsx:36]: image
image

File: src/components/Drawer.jsx (hunk 5 above)
Comment (SemenStruchev) [reply src/components/Drawer.jsx:16]: Rushi asked to add it

File: src/components/Drawer.jsx (hunk 3 above)
Comment (SemenStruchev) [reply src/components/Drawer.jsx:111]: Lint throw erroros without it

File: src/components/Drawer.jsx (hunk 4 above)
Comment (SemenStruchev) [reply src/components/Drawer.jsx:108]: Cos I don't think that this component'll be used without children

File: src/components/Drawer.jsx [hunk 6]
```diff
@@ -92,12 +92,12 @@ const CloseButton = ({ onClose }) => {
             size="small"
             variant="link"
             className={clsx(
-                "m-1 inline-flex !h-6 !w-6 items-center justify-center !rounded-full bg-white !px-1.5",
+                "m-[10px] inline-flex !h-10 !w-10 items-center justify-center !rounded-full bg-white !px-1.5",
```
Comment (manojx031) [inline src/components/Drawer.jsx:95]: can you change this to m-2.5

**Review by manojx031 — CHANGES_REQUESTED** [review]

**Review by manojx031 — APPROVED** [review]

File: src/components/Drawer.jsx (hunk 5 above)
Comment (rushi) [reply src/components/Drawer.jsx:16]: @manojx031 We are adding forwardRef on all of our primary components so that we can pass in refs from seller app to modify things. It's additional power for us. I've seen all of the UI libraries doing this.

File: src/components/Drawer.jsx (hunk 4 above)
Comment (rushi) [reply src/components/Drawer.jsx:108]: Agreed, Dialog should always have component

**Review by rushi — APPROVED** [review]
Pushed one minor change and approved.
